cypress/integration/models/stakeholdergroups.ts

```diff
@@ -47,100 +48,92 @@ export class Stakeholdergroups {
-            this.getStakeholdergroupDescription();
-            this.fillName(this.stakeholdergroupName);
-            this.fillDescription(this.stakeholdergroupDescription);
-            if (member) {
-                this.selectMember(member);
+            this.fillName(this.name);
+            this.fillDescription(this.description);
+            if (this.members) {
+                this.selectMembers(this.members);
             }
             submitForm();
             checkSuccessAlert(
-                successAlertMessage,
-                `Success! ${this.stakeholdergroupName} was added as a stakeholder group.`
+                commonView.successAlertMessage,
+                `Success! ${this.name} was added as a stakeholder group.`
             );
         }
     }
 
-    edit({
-        name = this.stakeholdergroupName,
-        description = this.stakeholdergroupName,
-        member = null,
-        cancel = false,
-    } = {}): void {
+    edit(
+        updatedValue: {
+            name?: string;
+            description?: string;
+            members?: Array<string>;
+        },
+        cancel = false
+    ): void {
         Stakeholdergroups.clickStakeholdergroups();
         selectItemsPerPage(100);
         cy.wait(2000);
         cy.get(tdTag)
-            .contains(this.stakeholdergroupName)
+            .contains(this.name)
             .parent(trTag)
             .within(() => {
-                click(editButton);
+                click(commonView.editButton);
             });
-
-        if (
-            !cancel &&
-            (name !== this.stakeholdergroupName ||
-                description !== this.stakeholdergroupDescription ||
-                member)
-        ) {
-            this.fillName(name);
-            this.fillDescription(this.stakeholdergroupDescription);
-            if (member) {
-                this.selectMember(member);
-            }
-            submitForm();
-            this.stakeholdergroupName = name;
-        } else {
+        if (cancel) {
             cancelForm();
+        } else {
+            if (updatedValue.name && updatedValue.name != this.name) {
+                this.fillName(updatedValue.name);
+                this.name = updatedValue.name;
+            }
+            if (updatedValue.description && updatedValue.description != this.description) {
+                this.fillDescription(updatedValue.description);
+                this.description = updatedValue.description;
+            }
+            if (updatedValue.members && updatedValue.members.length != 0) {
+                this.selectMembers(updatedValue.members);
+                this.members = updatedValue.members;
+            }
+            if (updatedValue) submitForm();
         }
     }
 
-    delete({ name = this.stakeholdergroupName, cancel = false } = {}): void {
+    delete(cancel = false): void {
         Stakeholdergroups.clickStakeholdergroups();
         selectItemsPerPage(100);
```

Comment:
can you please add name as optional argument to delete particular name?

Reply:
I want to re-iterate the correct use of object and class here. When you create an object for any of the model class, you need to have that object to perform specific tasks on it like - create, edit and delete. You cannot create just one object and use it to create multiple stakeholders or stakeholdergroups etc. Each object is unique and it has its task to be done. So, the question itself is wrong here ! If you have a object that was used to create any of the control, then same object will be used to call the delete on itself.